sdk/iOS/OoyalaSkinSDK/OOUpNextManager.m

```diff
@@ -82,7 +83,20 @@ - (void)goToNextVideo {
   // if upnext has not been dismissed and there is a next video, play the next video.
   LOG(@"Going to next video based on Up Next");
   if (!self.isDismissed && self.nextVideo) {
-    [self.player setEmbedCode:self.nextVideo[embedCodeKey]];
+    
+#warning: old API. Remove when SDK version > 4.46.0_GA. If you need to use old API, uncomment -playForJustChangedItem in -bridgeCurrentItemChangedNotification: of 'OOSkinPlayerObserver'
+    //[self.player setEmbedCode:self.nextVideo[embedCodeKey]];
+    
+    //new API from OOyalaSDK. Available > 4.46.0_GA
+    __weak OOUpNextManager *weakSelf = self;
+    [self.player setEmbedCode:self.nextVideo[embedCodeKey] withCallback:^(OOOoyalaError *error) {
+      NSLog(@"✅ Got callback. Is setEmbedCode successfull: [%d]", (error == nil));
```

Comment:
I guess it's better to use `true/false` or `YES/NO` instead of 0 and 1 :)

Reply:
And one more thing
The purpose of using `LOG` method from `OoayalaSDK` was in disabling them when in release.

`NSLog` here will print this debug message _always_ , and I don't think it's a good idea.

Reply:
Done